extension/js/common/core/smime.ts

```diff
@@ -16,3 +16,26 @@ export const encrypt = (pubkeys: string[], data: Uint8Array): { data: Uint8Array
   }
   return { data: new Uint8Array(arr), type: 'smime' };
 };
+
+export const getSerialNumber = (pubkey: string) => {
+  const certificate = forge.pki.certificateFromPem(pubkey);
+  return certificate.serialNumber;
+};
+
+export const getExpirationTime = (pubkey: string) => {
+  const certificate = forge.pki.certificateFromPem(pubkey);
+  const expiration = certificate.validity.notAfter;
```

Comment:
It's preferred to return a `Date` for dates, so in this case `Date | undefined` would be ideal